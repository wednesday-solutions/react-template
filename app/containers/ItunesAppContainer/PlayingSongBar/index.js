/**
 *
 * PlayingSongBar
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { PauseCircleOutlined } from '@ant-design/icons';
import If from '@app/components/If/index';

const CustomPauseButton = styled(PauseCircleOutlined)`
  && {
    color: red;
    font-size: 30px;
    cursor: pointer;
  }
`;
const Container = styled.div`
  && {
    display: flex;
    align-items: baseline;
  }
  .song {
    display: flex;
    flex-direction: column;
    margin-left: 10px;
  }
  .song-name {
    font-size: 20px;
  }

  .tracktype {
    text-transform: uppercase;
    font-size: 12px;
  }
`;
function PlayingSongBar({ song, dispatchPlayingSong }) {
  return (
    <Container data-testid="playing-song-bar">
      <div>
        <If condition={song}>
          <CustomPauseButton onClick={() => dispatchPlayingSong(null)} />
        </If>
      </div>
      <div className="song">
        <span className="song-name">{song?.trackName}</span>

        <span className="tracktype">
          <span>{song?.primaryGenreName}</span>:
          <span style={{ marginLeft: 10 }}>{new Date(song?.releaseDate).getFullYear()}</span>
        </span>
      </div>
    </Container>
  );
}

PlayingSongBar.propTypes = { song: PropTypes.Object, dispatchPlayingSong: PropTypes.func };

export default PlayingSongBar;
