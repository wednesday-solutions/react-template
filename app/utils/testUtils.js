import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter, Router } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';
import configureStore from '@app/configureStore';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@containers/LanguageProvider';
import { IntlGlobalProvider } from '@components/IntlGlobalProvider';
import { theme } from '@containers/App';

export const renderWithIntl = (children) =>
  render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );

export const renderProvider = (children, history) => {
  const store = configureStore({}, browserHistory).store;
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        <ThemeProvider theme={theme}>
          {history ? <Router history={history}>{children}</Router> : <BrowserRouter>{children}</BrowserRouter>}
        </ThemeProvider>
      </ConnectedLanguageProvider>
    </Provider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const getStyles = () => {};
