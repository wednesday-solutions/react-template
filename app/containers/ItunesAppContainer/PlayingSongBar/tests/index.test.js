/**
 *
 * Tests for PlayingSongBar
 *
 */

import React from 'react';
import { fireEvent } from '@testing-library/dom';
import { renderWithIntl } from '@utils/testUtils';
import PlayingSongBar from '../index';

describe('<PlayingSongBar />', () => {
  let dispatchPlayingSongSpy;
  beforeEach(() => {
    dispatchPlayingSongSpy = jest.fn();
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<PlayingSongBar />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 PlayingSongBar component', () => {
    const { getAllByTestId } = renderWithIntl(<PlayingSongBar />);
    expect(getAllByTestId('playing-song-bar').length).toBe(1);
  });

  it('should dispatch dispatchPlayingSong action', () => {
    const { getByTestId } = renderWithIntl(
      <PlayingSongBar song={{ trackId: 1, previewUrl: 'sampleurl' }} dispatchPlayingSong={dispatchPlayingSongSpy} />
    );
    fireEvent.click(getByTestId('pause_btn'));
    expect(dispatchPlayingSongSpy).toBeCalled();
  });
});
