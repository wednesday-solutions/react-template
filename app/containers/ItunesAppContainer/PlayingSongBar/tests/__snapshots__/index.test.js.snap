// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayingSongBar /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="PlayingSongBar__Container-sc-3br0fj-1 dsMBZY"
      data-testid="playing-song-bar"
    >
      <div />
      <div
        class="song"
      >
        <span
          class="song-name"
        />
        <span
          class="tracktype"
        >
          <span />
          :
          <span
            style="margin-left: 10px;"
          >
            NaN
          </span>
        </span>
      </div>
    </div>
  </div>
</body>
`;
